### PR TITLE
Remove excessive usage of `string_format`

### PR DESCRIPTION
**Pull request by BevapDin**
```SUMMARY: None```

Removes usage when calling functions that do string formatting on their own (e.g. `add_msg`).
Removes it when called on a string without any formatting specifiers.
Removes it when called with the format string "%s" (as it just passes the next argument through).

Fix some missing / incorrect translation code.

### CODE REVIEW

File: src/magic.cpp [hunk 1]
```diff
@@ -1268,15 +1269,10 @@ class spellcasting_callback : public uilist_callback
...
-        return _( string_format( "%d %s", moves, "moves" ) );
-    } else if( moves < to_moves<int>( 2_minutes ) ) {
-        return _( string_format( "%d %s", turns, "turns" ) );
-    } else if( moves < to_moves<int>( 2_hours ) ) {
-        return _( string_format( "%d %s", to_minutes<int>( turns * 1_turns ), "minutes" ) );
+    if( moves < to_moves<int>( 2_seconds ) ) {
+        return string_format( _( "%d moves" ), moves );
     } else {
-        return _( string_format( "%d %s", to_hours<int>( turns * 1_turns ), "hours" ) );
+        return to_string( time_duration::from_turns( moves / 100 ) );
```
Comment (KorGgenT) [inline src/magic.cpp]: does this round the time duration string like in the previous code, or does this give you an exact number?

Comment (BevapDin) [reply src/magic.cpp]: It yields a string with up to two units (e.g. "2 hours and 5 minutes" - remaining seconds are ignored here).

File: src/magic.cpp [hunk 2]
```diff
@@ -909,13 +909,14 @@ std::string spell::enumerate_targets() const
         return all_valid_targets[0];
     }
     std::string ret;
+    // @todo if only we had a function to enumerate strings and concatenate them...
```
Comment (Qrox) [inline src/magic.cpp]: `enumerate_as_string` in `output.h`?

Comment (BevapDin) [reply src/magic.cpp]: Yes, I was ironically referring to that one. As of now, I leave it to someone else to actually implement it.

**Pull request merged**